src/Common/src/System/IO/PathInternal.Windows.StringBuffer.cs

```diff
@@ -15,27 +15,40 @@ internal static partial class PathInternal
         /// </summary>
         internal static bool IsExtended(StringBuffer path)
         {
-            Debug.Assert(path != null);
-            return path.StartsWith(ExtendedPathPrefix);
+            // While paths like "//?/C:/" will work, they're treated the same as "\\.\" paths.
+            // Skipping of normalization will *only* occur if back slashes ('\') are used.
+            return path.Length >= DevicePrefixLength
+                && path[0] == '\\'
+                && (path[1] == '\\' || path[1] == '?')
+                && path[2] == '?'
+                && path[3] == '\\';
```

Comment:
So I guess the method is misnamed then? Based on your comments I'd assume a path like `//?/C:/` is considered *extended*, but not *normalized*, hence will not skip normalization.